lib/Sema/CSSimplify.cpp

```diff
@@ -5015,7 +5039,19 @@ ConstraintSystem::simplifyKeyPathApplicationConstraint(
     }
     return SolutionKind::Unsolved;
   };
-  
+
+  if (auto fixedRootTy =
+          getFixedTypeRecursive(rootTy, subflags, /*wantRValue=*/true)) {
+
+    if (fixedRootTy->isAnyObject()) {
```

Comment:
Having this block here might be too early, I think it should go somewhere here - https://github.com/apple/swift/pull/23820/files#diff-69306f7b0822cf52e8c0639307ad7407R5073 

It has to behave the same way as in keypath constraint, because `rootTy` might be a type variable, which makes this treaky because I'm not sure if it's possible to delay solving this constraint until at least root type is resolved. You might try and run tests to see if anything breaks (return unsolved if both `rootTy` and `kpRootTy` are type variables).

Reply:
Hmm I kept it early because we return `unsolved()` at the end. Anyway, I have made the change. It's also fixed an issue with duplicate diagnostic!

If it looks good to you, can you run the CI?

Reply:
Sure, this seems safe to me.